experimental/ast/context_nodes.go

```diff
@@ -75,18 +69,19 @@ func (c *Context) NewDeclDef(args DeclDefArgs) DeclDef {
 		args.Keyword, args.Type, args.Name, args.Returns,
 		args.Equals, args.Value, args.Options, args.Body, args.Semicolon)
 
-	ptr := c.decls.defs.New(rawDeclDef{
+	decl := wrapDeclDef(c, c.decls.defs.New(rawDeclDef{
 		name:    args.Name.raw,
 		equals:  args.Equals.raw,
+		value:   args.Value.raw,
 		options: args.Options.ptr,
+		body:    args.Body.ptr,
 		semi:    args.Semicolon.raw,
-	})
-	decl := wrapDecl[DeclDef](arena.Untyped(ptr), c)
+	}))
 
-	if args.Type != nil {
+	if args.Type.Nil() {
 		decl.SetType(args.Type)
 	} else {
-		decl.SetType(TypePath{Path: rawPath{args.Keyword.raw, args.Keyword.raw}.With(c)})
+		decl.SetType(TypePath{Path: rawPath{args.Keyword.raw, args.Keyword.raw}.With(c)}.AsAny())
```

Comment:
You could do this computation of `type` and `signature` before the call to wrap, and then just `return wrapDeclDef(...)`. Why move the `wrap` up before all of the inputs are computed?